string_calculator/string-calculator.js

```diff
@@ -0,0 +1,59 @@
+//-----------------------------------------------------------
+// Evaluates the expression string, returns the result.
+//-----------------------------------------------------------
+function eval(expression){
+    if(!expression) return 0;
+    let delimiters = getCustomDelimiters(expression);
+    let nums = getNums(expression, delimiters);
+    return nums.reduce((total, item) => total + item);
+}
+
+//------------------------------------------------------------
+// Parses the expression string to extract an array of custom
+// delimiters defined on the first line of the string.
+//
+//  Expected formats:
+//      Single one-char delimiter
+//          //#\n
+//      Single multi-char delimiter
+//          //[###]\n
+//      Multiple mult-char or single-char delimiters 
+//          //[#][###]...[DELIMITER-N]\n
+//------------------------------------------------------------
+function getCustomDelimiters(expression) {
+    let delimiters = [];
+    if(expression.startsWith('//')) {
```

Comment:
There's nothing wrong with this but if you "bail early" you can reduce an indentation in your method. If you don't want to do anything if the expression doesn't start with the characters you can :

```
function getCustomeDelimiters(expression) {
  if (!expression.startsWith('//')) return [];

  // proceed
}
```